src/Microsoft.Extensions.Options/IOptionsInitializer.cs

```diff
@@ -0,0 +1,15 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace Microsoft.Extensions.Options
+{
+    public interface IOptionsInitializer<TOptions> where TOptions : class
```

Comment:
:+1: Even if I still don't know how we can round up the whole story we tell customers about how to define their own options, I think this is awesome.